dbms/src/Encryption/MasterKey.h

```diff
@@ -0,0 +1,73 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Common/RandomData.h>
+#include <Common/TiFlashException.h>
+#include <Encryption/AESCTRCipher.h>
+#include <Encryption/EncryptionKey.h>
+#include <common/types.h>
+
+namespace DB
+{
+
+class MasterKey
+{
+public:
+    explicit MasterKey(String master_key_)
+        : master_key(std::move(master_key_))
+    {}
+
+    EncryptionKey generateEncryptionKey()
+    {
+        String plain_text = DB::random::randomString(EncryptionKey::KEY_LENGTH);
+        String cipher_text = plain_text;
+        DB::Encryption::Cipher(
+            0,
+            cipher_text.data(),
+            EncryptionKey::KEY_LENGTH,
+            master_key,
+            EncryptionMethod::Aes256Ctr,
+            nullptr,
+            true);
+        return EncryptionKey(cipher_text, plain_text, 0);
+    }
+
+    EncryptionKey decryptEncryptionKey(const String & exported)
+    {
+        RUNTIME_CHECK_MSG(exported.size() == EncryptionKey::EXPORT_KEY_LENGTH, "Invalid exported key length");
+        char method = exported[0];
+        RUNTIME_CHECK_MSG(method == static_cast<char>(EncryptionMethod::Aes256Ctr), "Invalid encryption method");
```

Comment:
add `exported_size()` / method to the error msg

Reply:
fixed https://github.com/pingcap/tiflash/pull/8627/commits/ec099b079511c7556a1223044d0195c56ff4563d